angular/src/app/file-field/file-field.component.ts

```diff
@@ -0,0 +1,133 @@
+import { Component, Input, SimpleChanges, OnChanges, Output, EventEmitter } from '@angular/core';
+import { AppField, FileField } from 'src/types';
+import { emptyField } from '../shared/fields';
+
+export const defaultFileUploader = (_files: File[]) => Promise.resolve([]);
+export const MAX_FILE_SIZE_BYTES = 10_485_759; // 10 MB - 1 Byte
+export const validFileTypesRecord = {
+    pdf: "application/pdf",
+    docx: "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+    doc: "application/msword",
+    png: "image/png",
+    jpg: "image/jpg",
+    jpeg: "image/jpeg",
+    gif: "image/gif",
+    svg: "image/svg+xml",
+    bmp: "image/bmp",
+    tiff: "image/tiff",
+};
+
+export const validFileTypes = Object.values(validFileTypesRecord);
+
+export enum InvalidFileReason {
+    Unsupported,
+    ExceededMaxBytesLimit,
+}
+
+export const filesReducer = (acc: { selected: File[]; invalid: InvalidFile[]; }, file: File) => {
+    const invalidReasons: InvalidFileReason[] = [];
+    if (file.size > MAX_FILE_SIZE_BYTES) {
+        invalidReasons.push(InvalidFileReason.ExceededMaxBytesLimit);
+    }
+    if (!validFileTypes.includes(file.type)) {
+        invalidReasons.push(InvalidFileReason.Unsupported);
+    }
+
+    if (invalidReasons.length) {
+        acc.invalid.push({ name: file.name, reasons: invalidReasons });
+    } else {
+        acc.selected.push(file);
+    }
```

Comment:
Minor stylistic issue: https://eslint.org/docs/rules/no-else-return  

I would have put a 'return acc' after the `acc.invalid.push` line, and then shown the acceptance criteria afterward at the bottom level of the expression.  That reduces the cyclomatic complexity and emphasizes the [guard](https://en.wikipedia.org/wiki/Guard_(computer_science)) quality of your validation checks.

But this is fine.  You don't have to change it.  Just a readability thing, and super-minor-nitpicky of me to notice in this case.